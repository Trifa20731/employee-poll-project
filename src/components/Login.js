import { useState } from "react";
import { connect } from "react-redux";
import { Button, Form } from "react-bootstrap";
import { handleSetAuthedUser } from "../actions/shared";
import { useNavigate } from "react-router-dom";

const Login = (props) => {
  const navigate = useNavigate();
  const [userText, setUserText] = useState("");
  const [passwordText, setPasswordText] = useState("");


  const isButtonDisable = () => userText === "" || passwordText === "";
  const isUserExist = () => props.users[userText] !== undefined;
  const isUserPasswordCorrect = () => props.users[userText].password === passwordText;

  const handleChange = (e, setText) => {
    const text = e.target.value;
    setText(text);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if(isUserExist()) {
      if (isUserPasswordCorrect()) {
        props.dispatch(handleSetAuthedUser(props.users[userText]));
        navigate("/home");
      } else {
        console.log("wrong password");
      }
    } else {
      console.log("unexist user.")
    }
    setUserText("");
    setPasswordText("");
  };

  return (
    <div>
      <h1>Employee Polls</h1>
      <h2>Log In</h2>
      <Form onSubmit={handleSubmit}>
        <Form.Group className="mb-3" controlId="formBasicEmail">
          <Form.Label>User</Form.Label>
          <Form.Control
            type="text"
            placeholder="Username"
            value={userText}
            onChange={(e) => handleChange(e, setUserText)}
          />
        </Form.Group>
        <Form.Group className="mb-3" controlId="formBasicPassword">
          <Form.Label>Password</Form.Label>
          <Form.Control
            type="password"
            placeholder="Password"
            value={passwordText}
            onChange={(e) => handleChange(e, setPasswordText)}
          />
        </Form.Group>
        <Button variant="primary" type="submit" disabled={isButtonDisable()}>
          Submit
        </Button>
      </Form>
    </div>
  );
};

const mapStateToProps = ({ authedUser, users }) => {
  return {
    authedUser,
    users
  }
};

export default connect(mapStateToProps)(Login);
