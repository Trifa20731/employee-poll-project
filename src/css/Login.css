.title{
  text-align: center;
  margin-top: 70px;
  margin-bottom: 30px;
}

.subtitle{
  text-align: center;
  margin-top: 30px;
  margin-bottom: 10px;
}

.logo{
  display: block;
  margin-left: auto;
  margin-right: auto;
  width: 20%;
}

.form{
  margin-left: auto;
  margin-right: auto;
  width: 60%;
}

.button {
  margin: auto;
  width: 20%;
}