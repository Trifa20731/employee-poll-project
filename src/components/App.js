import { React, useEffect, Fragment } from "react";
import { connect } from "react-redux";
import { handleInitialData } from "../actions/shared";
import { LoadingBar } from "react-redux-loading-bar";
import { Route, Routes, Navigate } from "react-router-dom";

import Login from "./login/Login";
import Navigation from "./Navigation";
import Dashboard from "./dashboard/Dashboard";
import LeaderBoard from "./leaderboard/LeaderBoard";
import NewQuestion from "./new_question/NewQuestion";
import Question from "./question/Question";
import PageNotFound from "./PageNotFound";

const App = (props) => {
  useEffect(() => {
    props.dispatch(handleInitialData());
  }, []);

  return (
    <Fragment>
      <LoadingBar />
      <div>
        { props.loading === true ? (
          <div>
            <Routes>
              <Route exact path="/" element={<Login />} />
              <Route path="/home" element={ <Navigate to="/" />} />
              <Route path="/leaderboard" element={ <Navigate to="/" />} />
              <Route path="/add" element={ <Navigate to="/" />} />
              <Route path="/questions/:id" element={ <Navigate to="/" />} />
              <Route path="*" element= {<PageNotFound />} />
            </Routes>
          </div>
        ) : (
          <div>
            <Navigation authedUser={props.authedUser} />
            <Routes>
              <Route path="/home" element={<Dashboard />} />
              <Route path="/leaderboard" element={<LeaderBoard />} />
              <Route path="/add" element={<NewQuestion />} />
              <Route path="/questions/:id" element={<Question/>} />
              <Route exact path="/" element={<Login />} />
            </Routes>
          </div>
        )}
      </div>
    </Fragment>
  );
};

const mapStateToProps = ({ authedUser }) => {
  return {
    authedUser,
    loading: authedUser === null,
  }
};

export default connect(mapStateToProps)(App);
