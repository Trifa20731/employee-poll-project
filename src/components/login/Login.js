import { useState } from "react";
import { connect } from "react-redux";
import { Button, Form } from "react-bootstrap";
import { handleSetAuthedUser } from "../../actions/shared";
import { useNavigate } from "react-router-dom";
import Logo from "../../images/school.png";
import "../../css/Login.css"

const Login = (props) => {
  const navigate = useNavigate();
  const [userText, setUserText] = useState("");
  const [passwordText, setPasswordText] = useState("");
  const [loginErrorMessage, setLoginErrorMessage] = useState("");


  const isButtonDisable = () => userText === "" || passwordText === "";
  const isUserExist = () => props.users[userText] !== undefined;
  const isUserPasswordCorrect = () => props.users[userText].password === passwordText;

  const handleChange = (e, setText) => {
    const text = e.target.value;
    setText(text);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if(isUserExist()) {
      if (isUserPasswordCorrect()) {
        props.dispatch(handleSetAuthedUser(props.users[userText]));
        navigate("/home");
      } else {
        console.log("wrong password");
        setLoginErrorMessage("wrong password");
      }
    } else {
      console.log("unexist user.");
      setLoginErrorMessage("unexist user.");
    }
    setUserText("");
    setPasswordText("");
  };

  return (
    <div>
      <h1 className="title">Employee Polls</h1>
      <img className="logo" src={Logo} alt="logo"/>
      <h2 className="subtitle">Log In</h2>
      <Form className="form" onSubmit={handleSubmit}>
        <Form.Group className="mb-3" controlId="formBasicEmail">
          <Form.Label >User</Form.Label>
          <Form.Control
            type="text"
            placeholder="Username"
            value={userText}
            onChange={(e) => handleChange(e, setUserText)}
            data-testid='username-input'
          />
        </Form.Group>
        <Form.Group className="mb-3" controlId="formBasicPassword">
          <Form.Label>Password</Form.Label>
          <Form.Control
            type="password"
            placeholder="Password"
            value={passwordText}
            onChange={(e) => handleChange(e, setPasswordText)}
            data-testid='password-input'
          />
          { 
            loginErrorMessage === "" 
              ? (
                <Form.Text></Form.Text>
              ) 
              : (
                <Form.Text>{loginErrorMessage}</Form.Text>
              )
          }
        </Form.Group>
        <Button 
          className="button" 
          variant="primary" 
          type="submit" 
          disabled={isButtonDisable()}
          
        >
          Submit
        </Button>
      </Form>
    </div>
  );
};

const mapStateToProps = ({ authedUser, users }) => {
  return {
    authedUser,
    users
  };
};

export default connect(mapStateToProps)(Login);
