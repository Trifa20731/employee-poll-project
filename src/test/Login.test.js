import { render } from '@testing-library/react';
import * as React from 'react';
import { Provider } from 'react-redux';
import { store } from '../index';
import { BrowserRouter, MemoryRouter } from "react-router-dom";
import App from '../components/App';
import Login from '../components/login/Login';

describe('Login Page Test', ()=>{
  it('will have all expected filed', () => {
    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>
    );

    const userLabel = component.getAllByText(/User/);
    expect(userLabel.length).toEqual(1);
    const passwordLabel = component.getAllByText(/Password/);
    expect(passwordLabel.length).toEqual(1);

    var userNameInput = component.getByTestId('username-input');
    var passwordInput = component.getByTestId('password-input');
    expect(userNameInput).toBeInTheDocument();
    expect(passwordInput).toBeInTheDocument();

    var submitButton = component.getByText('Submit')
    expect(submitButton).toBeInTheDocument();

  });

  it('will show the error message when enter wrong password', () => {

  });

  it('will show the error message when enter unexist username', () => {

  });
})