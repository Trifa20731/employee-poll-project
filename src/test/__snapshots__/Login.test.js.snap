// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test login 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div />
      <div>
        <div>
          <div>
            <h1
              class="title"
            >
              Employee Polls
            </h1>
            <img
              alt="logo"
              class="logo"
              src="school.png"
            />
            <h2
              class="subtitle"
            >
              Log In
            </h2>
            <form
              class="form"
            >
              <div
                class="mb-3"
              >
                <label
                  class="form-label"
                  for="formBasicEmail"
                >
                  User
                </label>
                <input
                  class="form-control"
                  data-testid="username-input"
                  id="formBasicEmail"
                  placeholder="Username"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="mb-3"
              >
                <label
                  class="form-label"
                  for="formBasicPassword"
                >
                  Password
                </label>
                <input
                  class="form-control"
                  data-testid="password-input"
                  id="formBasicPassword"
                  placeholder="Password"
                  type="password"
                  value=""
                />
                <small
                  class="form-text"
                />
              </div>
              <button
                class="button btn btn-primary"
                disabled=""
                type="submit"
              >
                Submit
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div />
    <div>
      <div>
        <div>
          <h1
            class="title"
          >
            Employee Polls
          </h1>
          <img
            alt="logo"
            class="logo"
            src="school.png"
          />
          <h2
            class="subtitle"
          >
            Log In
          </h2>
          <form
            class="form"
          >
            <div
              class="mb-3"
            >
              <label
                class="form-label"
                for="formBasicEmail"
              >
                User
              </label>
              <input
                class="form-control"
                data-testid="username-input"
                id="formBasicEmail"
                placeholder="Username"
                type="text"
                value=""
              />
            </div>
            <div
              class="mb-3"
            >
              <label
                class="form-label"
                for="formBasicPassword"
              >
                Password
              </label>
              <input
                class="form-control"
                data-testid="password-input"
                id="formBasicPassword"
                placeholder="Password"
                type="password"
                value=""
              />
              <small
                class="form-text"
              />
            </div>
            <button
              class="button btn btn-primary"
              disabled=""
              type="submit"
            >
              Submit
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
